klpbuild/ibs.py

```diff
@@ -120,11 +120,23 @@ def extract_rpms(self, args):
     def download_and_extract(self, args):
         i, cs, _, _, arch, _, rpm, dest = args
 
-        self.download_binary_rpms(args)
+        # Try to download and extract at least twice if any problems arise
+        tries = 2
+        while tries > 0:
+            self.download_binary_rpms(args)
+            try:
+                self.extract_rpms((i, cs, arch, rpm, dest))
+                # All good, stop the loop
+                break
+            except Exception as e:
+                # There was an issue when extracting the RPMs, probably because it's broken
+                # Remove the downloaded RPMs and try again
+                tries = tries - 1
+                logging.info(f"Problem to extract {rpm}. Downloading it again")
+                Path(dest, rpm).unlink()
 
-        # Do not extract kernel-macros rpm
-        if "kernel-macros" not in rpm:
-            self.extract_rpms((i, cs, arch, rpm, dest))
+        if tries == 0:
+            raise RuntimeError(f"Failed to download {rpm}. Aborting")
```

Comment:
s/download/extract/ no? To keep consistency with the prev log. Also, I think it could be useful to print here the `stderr` from the rpm2cpio process. What do you think?